java/src/test/java/com/saucelabs/simplesauce/SauceSessionTest.java

```diff
@@ -135,10 +136,43 @@ public void startThrowsErrorWithoutAccessKey() {
     }
 
     @Test
-    public void stop_noParams_callsDriverQuit() {
+    public void stopWithBooleanTrue() {
+        doReturn(dummyJSExecutor).when(sauce).getJSExecutor();
+        sauce.start();
+        sauce.stop(true);
+        verify(dummyJSExecutor).executeScript("sauce:job-result=passed");
+    }
+
+    @Test
+    public void stopWithBooleanFalse() {
+        doReturn(dummyJSExecutor).when(sauce).getJSExecutor();
+        sauce.start();
+        sauce.stop(false);
+        verify(dummyJSExecutor).executeScript("sauce:job-result=failed");
+    }
+
+    @Test
+    public void stopWithStringPassed() {
+        doReturn(dummyJSExecutor).when(sauce).getJSExecutor();
+        sauce.start();
+        sauce.stop("passed");
+        verify(dummyJSExecutor).executeScript("sauce:job-result=passed");
+    }
+
+    @Test
+    public void stopWithStringFailed() {
+        doReturn(dummyJSExecutor).when(sauce).getJSExecutor();
+        sauce.start();
+        sauce.stop("failed");
+        verify(dummyJSExecutor).executeScript("sauce:job-result=failed");
+    }
+
+    @Test
+    public void stopWithNoParameters() {
+        doReturn(dummyJSExecutor).when(sauce).getJSExecutor();
         sauce.start();
         sauce.stop();
 
-        verify(dummyRemoteDriver).quit();
+        verify(dummyJSExecutor, never()).executeScript(Mockito.anyString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Mockito.anyString' due to existing static import 'org.mockito.Mockito.*'](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4735013)